docs/branching-and-tagging.md

```diff
@@ -30,7 +30,7 @@ menu:
 
 ## Overview
 
-Iceberg table metadata maintains a log of snapshots which represent the changes applied to a table.
+Iceberg table metadata maintains a lot of snapshots which represent the changes applied to a table.
```

Comment:
I'm not sure this is a typo, I think this actually means that it's holding a snapshot log. To make this clearer maybe reword this to `Iceberg table metadata maintains a snapshot log, which represent the changes applied to a table.`

Reply:
@nastra Thanks for clarification. It seems that i misunderstood the sentence. :)

Reply:
Yeah it's a log of snapshots, but I like @nastra suggestion, we can use "snapshot log" so it's more clear. thanks @zhangbutao !

Reply:
I made a little change on this sentence according your guys suggestion. Thanks.